### DISCUSSION TITLE
way to retain the sign-ins (save the cookies)?

### DISCUSSION THREAD

**Discussion by ank-007** [CefView/QCefView#349] [General]
I want to know if it is possible to retain all the sign-ins don on the QCefview, 
for example, let's say I launch youtube on QCefView and sign-in with my account, I want it to save the acc info for the other QCefView too. is it possible to save my acc info in any way?
if I am able to save it even after i kill and rerun my Qt application , that is even great?

please help @tishion @le0zh  @ZhouYing-bit  @leashi  @OgreTransporter @L-Super

**Comment by L-Super** [comment]
You can use  [setCachePath()](https://cefview.github.io/QCefView/docs/reference/QCefConfig/#class_q_cef_config_1aa8f73284ec9ed73dc2028b8c89e8e3c8)  to set the cache path for saving cookies

**Reply by ank-007** [reply]
Thank you, this was helpful, I was able to retain sign-ins

**Reply by ank-007** [reply]
how to maintain multiple user profiles, is there a way change the cache path while my application is running? also is there a concept of incognito or a private tab kinda thing?

**Reply by L-Super** [reply]
> how to maintain multiple user profiles, is there a way change the cache path while my application is running? also is there a concept of incognito or a private tab kinda thing?

1. You can change the cache path by `setCachePath()`. If CEF can change the cache path while app is running, QCefView can
2. The default mode is stealth mode.

**Reply by ank-007** [reply]
found this [https://www.magpcss.org/ceforum/viewtopic.php?f=6&t=11695](https://www.magpcss.org/ceforum/viewtopic.php?f=6&t=11695), This is very similar to what I need, But how to create the subdomain, like he is mentioning. or is there way to change the cef path while my application is running.

for example, in my sample application, I setcachePath in main.cpp initially. but after some time if i want to change the userprofle, i want to change the cefCache path from the mainwindow for the same CefContext. is it possible?

**Comment by L-Super** [comment]
> found this https://www.magpcss.org/ceforum/viewtopic.php?f=6&t=11695, This is very similar to what I need, But how to create the subdomain, like he is mentioning. or is there way to change the cef path while my application is running.
> 
> for example, in my sample application, I setcachePath in main.cpp initially. but after some time if i want to change the userprofle, i want to change the cefCache path from the mainwindow for the same CefContext. is it possible?

You can try it, I haven't tried it.
